provider/pkg/provider/provider_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024, Pulumi Corporation.  All rights reserved.
+
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/pulumi/providertest"
```

Comment:
I think we may've wanted to deprecate that (@danielrbradley ?) and recommend using providertest/pulumitest for new work.

Reply:
... we should really update the main readme and deprecate the old methods.

Reply:
I rewrote the test using `PreviewProviderUpgrade` and it passes... but stack.json in testdata is also modified, and much shorter, although I didn't re-record??

Reply:
I think it records to a slightly different filename IICR so might have just auto-re-recorded first time. No idea why it would be shorter

Reply:
Pushed - could you sanity-check?

Reply:
Yup, looks good. Just need to delete the old `state.json` file.

Reply:
Pushed. Ready for a stamp.